tests/e2e/rosa_autoscaler_test.go

```diff
@@ -58,94 +51,182 @@ var _ = Describe("Autoscaler", labels.Feature.Autoscaler, func() {
+					Expect(err).ToNot(HaveOccurred())
+					logVerbosity := yamlData["log_verbosity"].(int)
+					balancingIgnoredLabels := yamlData["balancing_ignored_labels"].([]interface{})[0]
+					maxNodeProvisionTime := yamlData["max_node_provision_time"].(string)
+					maxPodGracePeriod := yamlData["max_pod_grace_period"].(int)
+					podPriorityThresold := yamlData["pod_priority_threshold"].(int)
+					coreMin := yamlData["resource_limits"].(map[string]interface{})["cores"].(map[string]interface{})["min"].(int)
+					coreMax := yamlData["resource_limits"].(map[string]interface{})["cores"].(map[string]interface{})["max"].(int)
+					memoryMin := yamlData["resource_limits"].(map[string]interface{})["memory"].(map[string]interface{})["min"].(int)
+					memoryMax := yamlData["resource_limits"].(map[string]interface{})["memory"].(map[string]interface{})["max"].(int)
+					maxNodesTotal := yamlData["resource_limits"].(map[string]interface{})["max_nodes_total"].(int)
+					sdDelayAfterAdd := yamlData["scale_down"].(map[string]interface{})["delay_after_add"].(string)
+					sdDelayAfterDelete := yamlData["scale_down"].(map[string]interface{})["delay_after_delete"].(string)
+					sdDelayAfterFailure := yamlData["scale_down"].(map[string]interface{})["delay_after_failure"].(string)
+					sdUtilizationThreshold := yamlData["scale_down"].(map[string]interface{})["utilization_threshold"].(string)
 
-				By("Describe autoscaler to check the edited value is correct")
-				rosaClient.Runner.YamlFormat()
-				yamlOutput, err = rosaClient.AutoScaler.DescribeAutoScaler(clusterID)
-				Expect(err).ToNot(HaveOccurred())
-				rosaClient.Runner.UnsetFormat()
-				yamlData, err = rosaClient.Parser.TextData.Input(yamlOutput).Parse().YamlToMap()
-				Expect(err).ToNot(HaveOccurred())
-				Expect(yamlData["ignore_daemonsets_utilization"]).To(Equal(true))
-				Expect(yamlData["resource_limits"].(map[string]interface{})["cores"].(map[string]interface{})["max"]).To(Equal(10))
-				Expect(yamlData["resource_limits"].(map[string]interface{})["gpus"].([]interface{})[0].(map[string]interface{})["range"].(map[string]interface{})["max"]).To(Equal(5))
-				Expect(yamlData["resource_limits"].(map[string]interface{})["gpus"].([]interface{})[0].(map[string]interface{})["range"].(map[string]interface{})["min"]).To(Equal(1))
-				Expect(yamlData["resource_limits"].(map[string]interface{})["gpus"].([]interface{})[0].(map[string]interface{})["type"]).To(Equal("amd.com/gpu"))
-				Expect(yamlData["scale_down"].(map[string]interface{})["delay_after_add"]).To(Equal("0s"))
-
-				By("Delete the autoscaler of the cluster")
-				resp, err = rosaClient.AutoScaler.DeleteAutoScaler(clusterID)
-				Expect(err).ToNot(HaveOccurred())
-				textData = rosaClient.Parser.TextData.Input(resp).Parse().Tip()
-				Expect(textData).To(ContainSubstring("INFO: Successfully deleted autoscaler configuration for cluster '%s'", clusterID))
+					By("Edit the autoscaler of the cluster")
```

Comment:
i think it would be better to put "edit/describe/delete" action in the common part.
if clusterConfig.Autoscaler != nil {
//Record the origin data 
//defer  
}else{
//create
}
//common part
//edit/describe/delete

Reply:
Done